### PR TITLE
Validation model n+1 problems

### PR DESCRIPTION
**Pull request by billionsjoel**
**Project requirements**

- Add the following validations:
- For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.
- Add flash messages in the create actions in all your controllers.
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,16 +1,16 @@
 class PostsController < ApplicationController
   def index
-    @user = current_user
-    @posts = current_user.last_three_posts
+    @user = User.includes(:posts).find(params[:user_id])
+    @page = params.fetch(:page, 0).to_i
+    @posts = @user.posts.offset(@page * POSTS_PER_PAGE).limit(POSTS_PER_PAGE)
```
Comment (ahmetbozaci) [inline app/controllers/posts_controller.rb:7]: - [ ] You already solve the n+1 problem in the show method. Also, do the same thing @posts variable.  Example: `@user.posts.includes(:comments)`
- [ ] Please fix the error when see all posts. 
 image
___

OPTIONAL
I wrote it as an optional but this is also required that has to do with the last milestone. so please fix it.
- [ ] ` See all posts` button is not working. You should fix it too.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -36,22 +37,35 @@ def update
...
+  def failed
+    flash.now[:alert] = 'Sorry, the post was not saved'
+  end
+
+  def show_errors
+    failed
+    errors = @post.errors.map(&:full_message)
+    flash.now[:error] = errors.join(' | ')
+  end
+
```
Comment (ahmetbozaci) [inline app/controllers/posts_controller.rb]: - [ ] Kindly fix the flash messages when  creating a post

image

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @billionsjoel ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._
- [ ] You should also create flash messages for the like controller. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec_n%2B1.md#:~:text=Add%20flash%20messages%20in%20the%20create%20actions%20in%20all%20your%20controllers.)

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏

almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by arslanbisharat — APPROVED** [review]
Hi,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**